public/components/ViewTypeToggle.vue

```diff
@@ -124,6 +135,23 @@ export default Vue.extend({
 
       return (hasLat && hasLon) || hasGeocoord;
     },
+    hasTrainingGeoVariables(): boolean {
```

Comment:
instead of `Array.filter().length > 0` you can use `Array.some` that returns a boolean.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
This is keeping consistent with hasGeoVariables. If we change hasTrainingGeoVariables. I would recommend changing the above function to.

Reply:
Please do.

We do not have proper coding style in this project, only Prettier.js. So if you see code that can be rewrite in a manner that is clearer, cleaner, or faster, update it.

We don't need to be consistent, apart of naming convention.

Reply:
Will do.